src/Util/Search.php

```diff
@@ -765,4 +779,60 @@ private function doesUdiExistInQueryTerm(string $queryTerm, Query\BoolQuery $fie
         }
         return $queryTerm;
     }
+
+    /**
+     * Split the query terms into must match and must not match terms.
+     *
+     * @param string $queryTerm Query term that needs to be searched upon.
+     *
+     * @return array
+     */
+    private function splitQueryTerms(string $queryTerm): array
+    {
+        $splitUpQueryTerms = array();
+        if (preg_match_all('/\s-\b(\w*)\b/', $queryTerm, $matches)) {
```

Comment:
Regex still a bit wrong, if you start your search with a negative item first, it will ignore the dash.